docs/src/literate-tutorials/ns_vs_diffeq.jl

```diff
@@ -380,62 +440,138 @@ function navierstokes!(du,u_uc,p,t)
-# We want to use the adaptive implicit Euler method with our custom linear
-# solver, which helps in the enforcement of the Dirichlet BCs. Further we
+# We want to use an adaptive variant of the implicit Euler method. Further we
 # enable the progress bar with the `progress` and `progress_steps` arguments.
 # Finally we have to communicate the time step length and initialization
 # algorithm. Since we start with a valid initial state we do not use one of
 # DifferentialEquations.jl initialization algorithms.
-# NOTE: At the time of writing this [no Hessenberg index 2 initialization is implemented](https://github.com/SciML/OrdinaryDiffEq.jl/issues/1019).
+# !!! note "DAE initialization"
+#     At the time of writing this [no Hessenberg index 2 initialization is implemented](https://github.com/SciML/OrdinaryDiffEq.jl/issues/1019).
 #
 # To visualize the result we export the grid and our fields
 # to VTK-files, which can be viewed in [ParaView](https://www.paraview.org/)
 # by utilizing the corresponding pvd file.
-timestepper = ImplicitEuler(linsolve = UMFPACKFactorization(reuse_symbolic=false))
+timestepper = Rodas5P(autodiff=false, step_limiter! = ferrite_limiter!);
+# timestepper = ImplicitEuler(nlsolve=NonlinearSolveAlg(OrdinaryDiffEq.NonlinearSolve.NewtonRaphson(autodiff=OrdinaryDiffEq.AutoFiniteDiff()); max_iter=50), step_limiter! = ferrite_limiter!)
+#NOTE!   This is left for future reference                                 #src
+# function algebraicmultigrid(W,du,u,p,t,newW,Plprev,Prprev,solverdata)   #src
+#     if newW === nothing || newW                                         #src
+#         Pl = aspreconditioner(ruge_stuben(convert(AbstractMatrix,W)))   #src
+#     else                                                                #src
+#         Pl = Plprev                                                     #src
+#     end                                                                 #src
+#     Pl,nothing                                                          #src
+# end                                                                     #src
+# timestepper = ImplicitEuler(linsolve = IterativeSolversJL_GMRES(; abstol=1e-8, reltol=1e-6), precs=algebraicmultigrid, concrete_jac=true) #src
+
+# !!! info "Debugging convergence issues"
+#     We can obtain some debug information from OrdinaryDiffEq by wrapping the following section into a [debug logger](https://docs.julialang.org/en/v1/stdlib/Logging/#Example:-Enable-debug-level-messages).
 integrator = init(
     problem, timestepper, initializealg=NoInit(), dt=Δt₀,
-    adaptive=true, abstol=1e-3, reltol=1e-3,
+    adaptive=true, abstol=1e-4, reltol=1e-5,
     progress=true, progress_steps=1,
-    saveat=Δt_save);
+    verbose=true, internalnorm=FreeDofErrorNorm(ch)
+);
+
 
 pvd = VTKFileCollection("vortex-street", grid);
-integrator = TimeChoiceIterator(integrator, 0.0:Δt_save:T)
-for (u_uc,t) in integrator
+for (u_uc,t) in TimeChoiceIterator(integrator, 0.0:Δt_save:T)
```

Comment:
```suggestion
for (u,t) in TimeChoiceIterator(integrator, 0.0:Δt_save:T)
```
And delete apply below. The solution should be computed correctly.